### PR TITLE
rename autogen to AG2 in website

### PR DESCRIPTION
**Pull request by skzhang1**
## Why are these changes needed?
rename autogen to AG2 in website

## Checks

- [x] I've included any doc changes needed for https://ag2ai.github.io/autogen/. See https://ag2ai.github.io/ag2/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

File: NOTICE.md [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 Copyright (c) 2023-2024, Owners of https://github.com/ag2ai
 
-This project is a fork of https://github.com/microsoft/autogen.
+This project is a fork of https://github.com/ag2ai/ag2.
```
Comment (sonichi) [inline NOTICE.md:5]: Remove this change

File: NOTICE.md [hunk 2]
```diff
@@ -2,9 +2,9 @@
...
 
-The [original project](https://github.com/microsoft/autogen) is licensed under the MIT License as detailed in [LICENSE_original_MIT](./license_original/LICENSE_original_MIT). The fork was created from version v0.2.35 of the original project.
+The [original project](https://github.com/ag2ai/ag2) is licensed under the MIT License as detailed in [LICENSE_original_MIT](./license_original/LICENSE_original_MIT). The fork was created from version v0.2.35 of the original project.
```
Comment (sonichi) [inline NOTICE.md:7]: Remove this change

File: NOTICE.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ This project, i.e., https://github.com/ag2ai/ag2, is licensed under the Apache L
 Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
 
 Ongoing MIT-licensed contributions:
-This project regularly incorporates code merged from the [original repository](https://github.com/microsoft/autogen) after the initial fork. This merged code remains under the original MIT license. For specific details on merged commits, please refer to the project's commit history.
-The MIT license applies to portions of code originating from the [original repository](https://github.com/microsoft/autogen) as described above.
+This project regularly incorporates code merged from the [original repository](https://github.com/ag2ai/ag2) after the initial fork. This merged code remains under the original MIT license. For specific details on merged commits, please refer to the project's commit history.
+The MIT license applies to portions of code originating from the [original repository](https://github.com/ag2ai/ag2) as described above.
```
Comment (sonichi) [inline NOTICE.md:17]: Remove these changes

File: README.md [hunk 4]
```diff
@@ -335,9 +335,9 @@ Explore detailed implementations with sample code and applications to help you g
 ## License
 This project is licensed under the [Apache License, Version 2.0 (Apache-2.0)](./LICENSE).
 
-This project is a spin-off of https://github.com/microsoft/autogen and contains code under two licenses:
+This project is a spin-off of https://github.com/ag2ai/ag2 and contains code under two licenses:
 
-- The original code from https://github.com/microsoft/autogen is licensed under the MIT License.  See the [LICENSE_original_MIT](./license_original/LICENSE_original_MIT) file for details.
+- The original code from https://github.com/ag2ai/ag2 is licensed under the MIT License.  See the [LICENSE_original_MIT](./license_original/LICENSE_original_MIT) file for details.
```
Comment (sonichi) [inline README.md:340]: Remove these changes

File: TRANSPARENCY_FAQS.md [hunk 5]
```diff
@@ -31,7 +31,7 @@ While AutoGen automates LLM workflows, decisions about how to use specific LLM o
...
-- We also evaluated [a team of AutoGen agents](https://github.com/microsoft/autogen/tree/gaia_multiagent_v01_march_1st/samples/tools/autogenbench/scenarios/GAIA/Templates/Orchestrator) on the [GAIA benchmarks](https://arxiv.org/abs/2311.12983), and got [SOTA results](https://huggingface.co/spaces/gaia-benchmark/leaderboard) as of
+- We also evaluated [a team of AutoGen agents](https://github.com/ag2ai/ag2/tree/gaia_multiagent_v01_march_1st/samples/tools/autogenbench/scenarios/GAIA/Templates/Orchestrator) on the [GAIA benchmarks](https://arxiv.org/abs/2311.12983), and got [SOTA results](https://huggingface.co/spaces/gaia-benchmark/leaderboard) as of
```
Comment (sonichi) [inline TRANSPARENCY_FAQS.md:34]: Remove this change

File: website/docs/installation/Docker.md [hunk 6]
```diff
@@ -1,37 +1,37 @@
...
 
-**Pre-configured DockerFiles**: The AutoGen Project offers pre-configured Dockerfiles for your use. These Dockerfiles will run as is, however they can be modified to suit your development needs. Please see the README.md file in autogen/.devcontainer
+**Pre-configured DockerFiles**: The AG2 Project offers pre-configured Dockerfiles for your use. These Dockerfiles will run as is, however they can be modified to suit your development needs. Please see the README.md file in autogen/.devcontainer
 
-- **ag2ai_base_img**: For a basic setup, you can use the `ag2ai_base_img` to run simple scripts or applications. This is ideal for general users or those new to AutoGen.
+- **ag2ai_base_img**: For a basic setup, you can use the `ag2ai_base_img` to run simple scripts or applications. This is ideal for general users or those new to AG2.
```
Comment (sonichi) [inline website/docs/installation/Docker.md:7]:
```suggestion
- **ag2_base_img**: For a basic setup, you can use the `ag2_base_img` to run simple scripts or applications. This is ideal for general users or those new to AG2.
```
Please double check if my suggestion is correct.

File: website/docs/installation/Docker.md [hunk 7]
```diff
@@ -1,37 +1,37 @@
...
 - **ag2ai_full_img**: Advanced users or those requiring more features can use `ag2ai_full_img`. Be aware that this version loads ALL THE THINGS and thus is very large. Take this into consideration if you build your application off of it.
```
Comment (sonichi) [inline website/docs/installation/Docker.md:8]:
```suggestion
- **ag2_full_img**: Advanced users or those requiring more features can use `ag2_full_img`. Be aware that this version loads ALL THE THINGS and thus is very large. Take this into consideration if you build your application off of it.
```
Please double check. If the docker file hasn't been renamed in other places, rename them to be consistent.

File: website/docs/installation/Docker.md [hunk 8]
```diff
@@ -1,37 +1,37 @@
...
   ```bash
   docker build -f .devcontainer/Dockerfile -t ag2ai_base_img https://github.com/ag2ai/ag2.git#main
   ```
 
-- **Autogen Advanced**: Advanced users or those requiring all the things that AutoGen has to offer `ag2ai_full_img`
+- **AG2 Advanced**: Advanced users or those requiring all the things that AG2 has to offer `ag2ai_full_img`
 
   ```bash
   docker build -f .devcontainer/full/Dockerfile -t ag2ai_full_img https://github.com/ag2ai/ag2.git#main
   ```
```
Comment (sonichi) [inline website/docs/installation/Docker.md:30]:
```suggestion
- **AG2 Advanced**: Advanced users or those requiring all the things that AG2 has to offer `ag2_full_img`

  ```bash
  docker build -f .devcontainer/full/Dockerfile -t ag2_full_img https://github.com/ag2ai/ag2.git#main
  ```
```

File: website/process_notebooks.py [hunk 9]
```diff
@@ -2,7 +2,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 #
-# Portions derived from  https://github.com/microsoft/autogen are under the MIT License.
+# Portions derived from  https://github.com/ag2ai/ag2 are under the MIT License.
```
Comment (sonichi) [inline website/process_notebooks.py:5]: Remove this change

File: website/src/components/ExploreContent/ExploreContent.js [hunk 10]
```diff
@@ -10,7 +10,7 @@ const firstDataRow = [
     link: "/docs/Getting-Started",
     description: (
       <>
-        Learn how to get start with AutoGen. Follow the instruction to quickly build-up your first AutoGen application.
+        Learn how to get start with AG2. Follow the instruction to quickly build-up your first AG2 application.
```
Comment (sonichi) [inline website/src/components/ExploreContent/ExploreContent.js:13]:
```suggestion
        Learn how to get started with AG2. Follow the instruction to quickly build-up your first AG2 application.
```

File: website/process_notebooks.py
Comment (sonichi) [inline website/process_notebooks.py:1]: More changes are required to render the notebook properly on the website. Could you test it locally first?

**Review by sonichi — CHANGES_REQUESTED** [review]

File: website/docs/installation/Docker.md (hunk 6 above)
Comment (skzhang1) [reply website/docs/installation/Docker.md:7]: checked. it is correct.

File: website/docs/installation/Docker.md (hunk 7 above)
Comment (skzhang1) [reply website/docs/installation/Docker.md:8]: Thanks! checked. it is correct.

File: website/docs/installation/Docker.md (hunk 8 above)
Comment (skzhang1) [reply website/docs/installation/Docker.md:30]: revised

File: website/process_notebooks.py
Comment (skzhang1) [reply website/process_notebooks.py:1]: It works well locally. Can you share which part is not properly rendered? Let me try to fix it.

Comment (sonichi) [reply website/process_notebooks.py:1]: For example, https://ag2ai.github.io/autogen/docs/notebooks/JSON_mode_example shows 404. All of the notebook pages show 404.

Comment (skzhang1) [reply website/process_notebooks.py:1]: resolved!

**Comment by skzhang1** [comment]
@sonichi we can merge this one to fix urgent issues. Fix others in later PR.

**Pull request merged**